File: src/app/api/responses/skill-response.ts
import {FlagResponse} from './flag-response';
import {StatModificationOperand} from '../shared/enums';

export interface SkillResponse {
    id: number;
    name: string;
    description?: string;
    playerDescription?: string;
    require?: string;
    resist?: string;
    using?: string;
    roleplay?: string;
    stat?: string[];
    test?: number;
    flags?: FlagResponse[];
    effects: {
        stat: string;
        value: number;
        type: StatModificationOperand;
    }[];
}

File: backend/src/Naheulbook.Web/Responses/SkillResponse.cs
using JetBrains.Annotations;

namespace Naheulbook.Web.Responses;

[PublicAPI]
public class SkillResponse
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public string? PlayerDescription { get; set; }
    public string? Require { get; set; }
    public string? Resist { get; set; }
    public string? Using { get; set; }
    public string? Roleplay { get; set; }
    public string[] Stat { get; set; } = null!;
    public short? Test { get; set; }
    public List<FlagResponse>? Flags { get; set; }
    public List<SkillEffectResponse> Effects { get; set; } = null!;
}

[PublicAPI]
public class SkillEffectResponse
{
    public string Stat { get; set; } = null!;
    public int Value { get; set; }
    public string Type { get; set; } = null!;
}